### PR TITLE
Prepare Backup Metrics for postgres

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by reimannf** [comment]
I think you miss the service for exposing the metrics

File: postgres/templates/service.yaml [hunk 1]
```diff
@@ -24,3 +24,28 @@ spec:
...
+    prometheus.io/scrape: "true"
+    prometheus.io/port: "9188"
+spec:
+  clusterIP: None
+  ports:
+    - name: backup-metrics
+      port: 9188
+      protocol: TCP
+  selector:
+    app: postgres-{{.Values.name}}
```
Comment (reimannf) [inline postgres/templates/service.yaml]: there is no label `app`. only `name`

**Review by reimannf — CHANGES_REQUESTED** [review]
Please change

**Comment by fwiesel** [comment]
I would suggest to combine this change with
- Updating postgres to a later teeny version (to fix the recent CVE security issue)
- Removing the postgres-metrics container from the deployment.

Is it required to run the backup in the same pod, or could we move it to another?
I am asking, as each update of the backup will result in a restart of the database.

**Comment by ruvr** [comment]
I would suggest to leave the metrics container in to ease detection of a database being down with prometheus. And one could also use the metrics container to gather static metrics that are not service specific (like db size).
Just the service specific metrics should come from the separate pg_metrics pod IMHO.

**Comment by reimannf** [comment]
@fwiesel: Michael and myself decided to push out that change before decoupling the backup from the database (https://issues.wdf.sap.corp/browse/CCM-1272). That one includes bugfixes and metrics.
Do we have plans to update postgres in the near future?

**Pull request merged**